src/modules/auth/hooks/useAuthActions.ts

```diff
@@ -0,0 +1,127 @@
+import { useState, useCallback } from 'react';
+
+import { CookieName, CookiesConfig } from '@/config/cookies';
+import { AddressUtils, IPermission } from '@/modules/core';
+import { SignWebAuthnPayload, TypeUser } from '../services';
+
+export type SingleAuthentication = {
+  avatar: string;
+  account: string;
+  accountType: TypeUser;
+  userId: string;
+  workspace: string;
+  webAuthn?: Omit<SignWebAuthnPayload, 'challenge'>;
+};
+
+export type WorkspaceAuthentication = {
+  workspace: string;
+  permissions: IPermission;
+};
+
+export type IUseAuthActionsState = {
+  avatar: string;
+  account: string;
+  accountType?: TypeUser;
+  userId: string;
+  workspaces: { single: string; current: string };
+  isInvalidAccount: boolean;
+  formattedAccount: string;
+  permissions?: IPermission;
+  webAuthn?: Omit<SignWebAuthnPayload, 'challenge'>;
+};
+
+export type IUseAuthActionHandler = {
+  logout: () => void;
+  setInvalidAccount: (isInvalid: boolean) => void;
+  singleAuthentication: (params: SingleAuthentication) => void;
+  workspaceAuthentication: (params: WorkspaceAuthentication) => void;
+  workspaceAuthenticationSingle: () => void;
+};
+
+export type IUseAuthActions = IUseAuthActionsState & IUseAuthActionHandler;
+
+const useAuthActions = (): IUseAuthActions => {
+  const [state, setState] = useState<IUseAuthActionsState>({
```

Comment:
Replace this state to an query by reactquery:
- create a new hook to consume a endpoint on API `/user/me/info` [more info](https://github.com/infinitybase/bako-safe-api/pull/193), you will create an method on service, and a hook `useUserInfo`, call on `useWorkspace`.
- move function `hasWallet` on `useAuth` to the same hook -> `useUserInfo`